### PR TITLE
feat(date-picker): add min and max attributes (VIV-1273)

### PR DESCRIPTION
**Pull request by RichardHelm**

### CODE REVIEW

File: libs/components/src/lib/date-picker/date-picker.spec.ts [hunk 1]
```diff
@@ -253,6 +327,42 @@ describe('vwc-date-picker', () => {
...
 			expect(textField.errorText).toBe('');
 		});
+
+		it('should move focus to the tabbable element when pressing tab in the text-field', async () => {
+			await openPopup();
+
+			pressKey('Tab');
+
+			expect(element.shadowRoot!.activeElement).toBe(
+				getDateButton('2023-08-10')
```
Comment (YonatanKra) [inline libs/components/src/lib/date-picker/date-picker.spec.ts:337]: Would that be more understandable?
```suggestion
                        const firstTabbableElement = getDateButton('2023-08-10');
			expect(element.shadowRoot!.activeElement).toBe(firstTabbableElement);
```

File: libs/components/src/lib/date-picker/date-picker.spec.ts [hunk 2]
```diff
@@ -253,6 +327,42 @@ describe('vwc-date-picker', () => {
...
+		});
+
+		it('should keep default behaviour when pressing tab in the text-field without a tabbable month', async () => {
+			const event = new KeyboardEvent('keydown', { key: 'Tab', bubbles: true });
+			event.preventDefault = jest.fn();
+			element.min = '2024-01-01';
+			element.value = '2023-01-01';
+			await openPopup();
+			titleAction.click();
+			await elementUpdated(element);
```
Comment (YonatanKra) [inline libs/components/src/lib/date-picker/date-picker.spec.ts:360]: Can this be extracted to `openMonthView()`?

File: libs/components/src/lib/date-picker/date-picker.spec.ts [hunk 3]
```diff
@@ -458,6 +560,18 @@ describe('vwc-date-picker', () => {
...
+
+		it('should remain on the current date when attempting to move into a date outside valid range', async () => {
+			element.max = '2023-08-01';
+			await elementUpdated(element);
+			getDateButton('2023-08-01').focus();
+
+			pressKey('ArrowRight');
+
+			expect(element.shadowRoot!.activeElement).toBe(
+				getDateButton('2023-08-01')
```
Comment (YonatanKra) [inline libs/components/src/lib/date-picker/date-picker.spec.ts:572]: ### 
```suggestion
			const focusedButton = getDateButton('2023-08-01');
			focusedButton.focus();

			pressKey('ArrowRight');

			expect(element.shadowRoot!.activeElement).toBe(focusedButton);
```

**Review by YonatanKra — CHANGES_REQUESTED** [review]
Some small remarks. Mostly about the tests.

File: libs/components/src/lib/date-picker/calendar/dateStr.ts [hunk 4]
```diff
@@ -20,6 +20,12 @@ export const addDays = (dateStr: DateStr, days: number): DateStr => {
 	return formatDateStr(resultDate);
 };
 
+export const compareDateStr = (a: DateStr, b: DateStr): number => {
+	const dateA = parseDateStr(a);
+	const dateB = parseDateStr(b);
+	return dateA.getTime() - dateB.getTime();
```
[the remark this answers is not in the document]
Comment (RichardHelm) [reply libs/components/src/lib/date-picker/calendar/dateStr.ts]: Sure, done 👍

File: libs/components/src/lib/date-picker/date-picker.spec.ts [hunk 5]
```diff
@@ -204,6 +204,80 @@ describe('vwc-date-picker', () => {
...
+	describe('min', () => {
+		beforeEach(async () => {
+			element.min = '2023-08-05';
+			await elementUpdated(element);
+			await openPopup();
+		});
+
+		describe('in the calendar grid', () => {
+			it('should disable dates earlier than min', async () => {
+				expect(getDateButton('2023-08-01').disabled).toBe(true);
```
[the remark this answers is not in the document]
Comment (RichardHelm) [reply libs/components/src/lib/date-picker/date-picker.spec.ts]: Sure, updated the tests to check all buttons now 👍

File: libs/components/src/lib/date-picker/date-picker.spec.ts [hunk 6]
```diff
@@ -204,6 +204,80 @@ describe('vwc-date-picker', () => {
...
+			});
+
+			it('should disable previous month button if the previous month is earlier than min', async () => {
```
[the remark this answers is not in the document]
Comment (RichardHelm) [reply libs/components/src/lib/date-picker/date-picker.spec.ts:241]: Added tests for that now

File: libs/components/src/lib/date-picker/date-picker.spec.ts (hunk 1 above)
Comment (RichardHelm) [reply libs/components/src/lib/date-picker/date-picker.spec.ts]: Yes, done

**Review by YonatanKra — APPROVED** [review]
LGTM! 
Nice job :)

**Pull request merged**